File: src/main/java/io/github/tanguygab/tabadditions/shared/features/advancedconditions/StringCondition.java
package io.github.tanguygab.tabadditions.shared.features.advancedconditions;

import me.neznamy.tab.shared.platform.TabPlayer;

import java.util.function.BiFunction;

public class StringCondition extends SimpleCondition {

    private final BiFunction<String, String, Boolean> function;

    public StringCondition(String[] arr, BiFunction<String, String, Boolean> function) {
        super(arr);
        this.function = function;
    }

    @Override
    public boolean isMet(TabPlayer viewer, TabPlayer target) {
        return function.apply(parseLeftSide(viewer,target), parseRightSide(viewer,target));
    }
}

File: src/main/kotlin/io/github/tanguygab/tabadditions/shared/features/advancedconditions/StringCondition.kt
package io.github.tanguygab.tabadditions.shared.features.advancedconditions

import me.neznamy.tab.shared.platform.TabPlayer

class StringCondition(arr: List<String>, private val function: (String, String) -> Boolean) : SimpleCondition(arr) {
    override fun isMet(viewer: TabPlayer, target: TabPlayer) = function(parseLeftSide(viewer, target), parseRightSide(viewer, target))
}
